include/swift/Reflection/TypeRef.h

```diff
@@ -307,10 +325,17 @@ class BoundGenericTypeRef final : public TypeRef, public NominalTypeTrait {
   create(Allocator &A, const std::string &MangledName,
          std::vector<const TypeRef *> GenericParams,
          const TypeRef *Parent = nullptr) {
+  
+    // GenericParams contains a flat vector of parameters used in the whole
+    // typeref hierarchy, so we skip the remove that belong to our parents.
+    if (Parent)
+      GenericParams.erase(GenericParams.begin(),
+                          GenericParams.begin() +
+                              GetNumberOfGenericParametersInHierarchy(Parent));
```

Comment:
Can you add a bounds check here to fail gracefully if `GetNumberOfGenericParametersInHierarchy(Parent) > GenericParams.size()`? We sometimes get weird values due to corrupt or misidentified data in the target process.